src/api/telemetry/TelemetryAPI.js

```diff
@@ -368,56 +413,52 @@ export default class TelemetryAPI {
     }
 
     /**
-     * Subscribe to realtime telemetry for a specific domain object.
-     * The callback will be called whenever data is received from a
-     * realtime provider.
+     * Request telemetry staleness for a domain object.
      *
-     * @method subscribe
+     * @method isStale
      * @memberof module:openmct.TelemetryAPI~TelemetryProvider#
      * @param {module:openmct.DomainObject} domainObject the object
-     *        which has associated telemetry
-     * @param {Function} callback the callback to invoke with new data, as
-     *        it becomes available
-     * @returns {Function} a function which may be called to terminate
-     *          the subscription
+     *        which has associated telemetry staleness
+     * @returns {Promise.<object[]>} a promise for a staleness data
+     *  object: ex.
+     *  {
+     *      isStale: <Boolean>,
+     *      timestamp: <timestamp>
+     *  }
      */
-    subscribe(domainObject, callback, options) {
-        const provider = this.#findSubscriptionProvider(domainObject);
+    async isStale(domainObject) {
+        const abortController = new AbortController();
+        const options = { signal: abortController.signal };
 
-        if (!this.subscribeCache) {
-            this.subscribeCache = {};
-        }
+        this.requestAbortControllers.add(abortController);
 
-        const keyString = objectUtils.makeKeyString(domainObject.identifier);
-        let subscriber = this.subscribeCache[keyString];
+        const provider = this.#findStalenessProvider(domainObject);
+        if (!provider) {
+            this.requestAbortControllers.delete(abortController);
+            const timeSystem = this.openmct.time.timeSystem();
 
-        if (!subscriber) {
-            subscriber = this.subscribeCache[keyString] = {
-                callbacks: [callback]
+            return {
+                [timeSystem.key]: 0,
```

Comment:
What are the implications of returning `undefined` here? I realize it's probably a bunch of extra boilerplate. Could we use optional chaining to make it less painful?

My thinking is, we don't actually know whether or not the telemetry point is stale, so returning `false` is technically incorrect, even if we don't currently differentiate between "unknown" and "not stale" in our UI.

Reply:
No, that's a good point. It's not too much work, I'll just return undefined here and not process anything if nothing is returned.

Reply:
Welp... that wasn't bad at all.